### PR TITLE
Goals page api

### LINKED ISSUE #171: Integration of GET API call to get all the goals in Goals App
I would like to request the integration of a GET API call in the Goals App to get all the goals at one place.

**BASE_URL** : https://api.realdevsquad.com

**API Endpoint** : {{BASE_URL}}/goal/

**NOTE**:  Please ensure that the necessary authentication and authorization mechanisms are in place to secure the API endpoint and protect user data.  For the auth please refer to the Postman collection link  
 https://api.postman.com/collections/18992903-eefca41f-d1e9-40dc-8dfa-b17920badb79?access_key=PMAT-01GZBS0MW8KH0VC5SWWSWK601S
 
 Ping in the channel in case of any confusion.

### PR DESCRIPTION
**Pull request by shreya-mishra**
Integrated Goals Page API through which we get the todos to show on the Goals page.

API: https://backend-goals-production.up.railway.app/goal/

### CODE REVIEW

File: src/components/ToDoComponent/TodoComponent.tsx [hunk 1]
```diff
@@ -6,14 +6,20 @@ import { TodoStyles } from './Styles/TodoStyles';
...
   const [disabled, setDisabled] = useState<boolean>(false);
   const [changed, setChanged] = useState<boolean>(false);
 
   useEffect(() => {
-    setTasks(tasks);
+    getTodos()
   }, [changed, tasks]);
 
+  const getTodos = async() => {
+    const todos = await fetch('https://backend-goals-production.up.railway.app/goal/');
```
Comment (sahsisunny) [inline src/components/ToDoComponent/TodoComponent.tsx]: Can you create the constants for this API URL fetch is default function or we created using axios or other library ?

File: src/components/ToDoComponent/TodoComponent.tsx [hunk 2]
```diff
@@ -6,14 +6,20 @@ import { TodoStyles } from './Styles/TodoStyles';
...
+    const todosJsonData = await todos.json();
+    console.log('todos',todosJsonData.data)
```
Comment (sahsisunny) [inline src/components/ToDoComponent/TodoComponent.tsx]: Why console.log not console.error ?

**Review by sahsisunny — COMMENTED** [review]
Can you change the all console.log to console.error ?

File: src/components/ToDoComponent/TodoComponent.tsx (hunk 2 above)
Comment (shreya-mishra) [reply src/components/ToDoComponent/TodoComponent.tsx]: this is general log ! we will be removing the same

File: src/components/ToDoComponent/TodoComponent.tsx (hunk 1 above)
Comment (shreya-mishra) [reply src/components/ToDoComponent/TodoComponent.tsx]: already done in previous PR will take a pull of the same and update it

**Comment by shreya-mishra** [comment]
> Can you change the all console.log to console.error ?

will you please let me know the usecase of the same ?

**Comment by sahsisunny** [comment]
> > Can you change the all console.log to console.error ?
> 
> will you please let me know the usecase of the same ?

Console.log is generally used for logging and debugging purposes it prints all types of objects and console.error is used for printing the error messages only error messages.

File: src/components/ToDoComponent/TodoComponent.tsx (hunk 1 above)
Comment (sahsisunny) [reply src/components/ToDoComponent/TodoComponent.tsx]: Ok

**Pull request merged**